### ISSUE TITLE
Clarify parameter-list-wrapping in documentation

### ISSUE THREAD

**Issue by Azamat989** [pinterest/ktlint#2080]
## **Observed behavior**

This function is formatted by ktlint like:
```kotlin
    fun setDimen(
        @DimenRes dimen: Int,
    )
```

## **Expected behavior**
But I expect singline formatting (because there is no violation of max-line-length rule and the ktlint_standard_annotation  rule is disabled):

```kotlin
    fun setDimen(@DimenRes dimen: Int)
```

## **Environment**

* Version of ktlint used: 0.49.1

* editorconfig:
```
ktlint_code_style = ktlint_official
ktlint_experimental=enabled
ktlint_standard_multiline-expression-wrapping = disabled
ktlint_standard_no-empty-first-line-in-class-body = disabled
ktlint_standard_function-signature = disabled
ktlint_standard_indent = disabled
ktlint_standard_package-name = disabled
ktlint_standard_annotation = disabled
ktlint_standard_max-line-length = disabled
ktlint_standard_no-wildcard-imports = disabled
ktlint_standard_string-template-indent = disabled
ktlint_ignore_back_ticked_identifier = true
```

* integrated with [jeremymailen/kotlinter-gradle](https://github.com/jeremymailen/kotlinter-gradle)

**Comment by paul-dingemans** [comment]
In the `ktlint_official` code style the annotated parameters are always wrapped in order to be more consistent with the annotation formatting.

For each code style in Ktlint it is assumed that all rules in that code style are enabled. When disabling certain rules, like you did with `ktlint_standard_annotation`, the end result might be actually be more inconsistent. Ktlint works best when you accept all defaults. The more you deviate from that default, the bigger the chance that things are not working as might be expected. Please recall that the original idea behind ktlint is to offer a way to lint/format code without too much configuration.

The documentation of the [parameter-list-rule](https://pinterest.github.io/ktlint/0.49.1/rules/standard/#parameter-wrapping) should be clarified.
